File: client/Assets/Scripts/Message/info/chenliang/moba/message/PlayerToBattleStub.cs
namespace info.chenliang.moba.message{
using System;
using System.IO;
using MiscUtil;
using MiscUtil.IO;
using MiscUtil.Conversion;
using Talky;
public abstract class PlayerToBattleStub {
protected abstract MemoryEndianBinaryWriter Begin();
protected abstract void End();
public void playerPing(){
MemoryEndianBinaryWriter writer = Begin();
writer.WriteByte(0);
End();
}
public void playerPong(){
MemoryEndianBinaryWriter writer = Begin();
writer.WriteByte(1);
End();
}
public void play(){
MemoryEndianBinaryWriter writer = Begin();
writer.WriteByte(2);
End();
}
public void moveTo(Single x,Single z){
MemoryEndianBinaryWriter writer = Begin();
writer.WriteByte(3);
writer.WriteSingle(x);
writer.WriteSingle(z);
End();
}
}
}

File: server/battle/src/main/java/info/chenliang/moba/message/PlayerToBattleStub.java
package info.chenliang.moba.message;
import java.io.*;
import info.chenliang.talky.*;
public abstract class PlayerToBattleStub {
protected abstract DataOutputStream begin();
protected abstract void end();
public void playerPing() throws Exception{
DataOutputStream dos = begin();
dos.write(0);
end();
}
public void playerPong() throws Exception{
DataOutputStream dos = begin();
dos.write(1);
end();
}
public void play() throws Exception{
DataOutputStream dos = begin();
dos.write(2);
end();
}
public void moveTo(float x,float z) throws Exception{
DataOutputStream dos = begin();
dos.write(3);
dos.writeFloat(x);
dos.writeFloat(z);
end();
}
}
